ctapipe/irf/irf_classes.py

```diff
@@ -117,28 +97,75 @@ def optimise_gh_cut(self, signal, background, alpha, max_bg_radius):
             op=operator.ge,
             theta_cuts=theta_cuts,
             alpha=alpha,
-            fov_offset_max=max_bg_radius * u.deg,
+            fov_offset_max=max_fov_radius * u.deg,
+            fov_offset_min=min_fov_radius * u.deg,
         )
 
         # now that we have the optimized gh cuts, we recalculate the theta
         # cut as 68 percent containment on the events surviving these cuts.
-        self.log.info("Recalculating theta cut for optimized GH Cuts")
         for tab in (signal, background):
             tab["selected_gh"] = evaluate_binned_cut(
                 tab["gh_score"], tab["reco_energy"], gh_cuts, operator.ge
             )
+        self.log.info("Recalculating theta cut for optimized GH Cuts")
 
-        theta_cuts = calculate_percentile_cut(
+        theta_cuts = psf.calculate_theta_cuts(
             signal[signal["selected_gh"]]["theta"],
             signal[signal["selected_gh"]]["reco_energy"],
             self.reco_energy_bins(),
-            percentile=68,
-            min_value=self.theta_min_angle * u.deg,
-            max_value=self.theta_max_angle * u.deg,
+        )
+
+        return gh_cuts, theta_cuts, sens2
+
+
+class PointSpreadFunction(Component):
```

Comment:
I'd not call this "PointSpreadFunction", since it is not a PointSpreadFunction, which is component of the IRF, it computes the theta cuts, which of course depend on the PSF, but it's not the same.